### ISSUE TITLE
Incorrect limits inside the container – shows hosts full memory and cpu cores

### ISSUE THREAD

**Issue by arren-ru** [lxc/lxc#3859]
# Required information

 * Distribution: Archlinux (systemd 248)
 * The output of
   * `lxc-start --version`
     ```
     4.0.9
     ```
   * [`lxc-checkconfig`](https://gist.githubusercontent.com/arren-ru/d1194dc24dec98f17d3e932c25856084/raw/85299048188d4f0b7dcc01cb6157c30150b87ac6/lxc-checkconfig)
   * `uname -a`
       ```
        Linux helios 5.12.9-arch1-1 #1 SMP PREEMPT Thu, 03 Jun 2021 11:36:13 +0000 x86_64 GNU/Linux
       ```
   * `cat /proc/self/cgroup` 0::/user.slice/user-0.slice/session-31.scope
   * [`cat /proc/1/mounts`](https://gist.githubusercontent.com/arren-ru/d1194dc24dec98f17d3e932c25856084/raw/1df0c6d183553b73be4cdc14e22f3868a8dcbd84/cat-proc-1-mounts)

# Issue description

I have LXC installed along with LXD. Creating containers with LXD gives normal containers with correct limits inside visible by `free` and `nproc` tools, but creating containers with `lxc-create` (choosing the same OS Archlinux as on the host) gives containers with limits equals to host one.
So basically, I can see in `lxc-create`-ed containers a full host memory and cpu set.
[config](https://gist.githubusercontent.com/arren-ru/d1194dc24dec98f17d3e932c25856084/raw/e5460a1d37cf6a1e1543029a6c9d3d5a46ab5579/config)

Digging around I found that this maybe a problem with LXCFS, so switched system to cgroup v1 mode with kernel parameter `systemd.unified_cgroup_hierarchy=0` and manually created containers are become ok.

So, the question is why and how to get normal behavior and limits with cgroup-v2, because LXD does it.

Sorry, if the issue is unrelated to this project, I could move it there if any.
Any help appreciated.

**Comment by brauner** [comment]
I'm confused. Do your containers created via `lxc-create` have any limits set? If you want to set limits for the unified cgroup hierarchy you need to use `lxc.cgroup2.*` keys. So it's not unlikely that your low-level LXC containers simply aren't using `lxc.cgroup2.*` keys.

**Comment by stgraber** [comment]
You're using lxc.cgroup instead of lxc.cgroup2 in your config, that's likely the issue

**Comment by arren-ru** [comment]
> I'm confused. Do your containers created via `lxc-create` have any limits set? If you want to set limits for the unified cgroup hierarchy you need to use `lxc.cgroup2.*` keys. So it's not unlikely that your low-level LXC containers simply aren't using `lxc.cgroup2.*` keys.

> You're using lxc.cgroup instead of lxc.cgroup2 in your config, that's likely the issue

Yes, that's the point, I personally have migrated from LXC3 to LXD in early versions and just didn't know about keys changed for low-level configuration, so that is my fault, sorry. LXD does job well to hide some details and make comfortable configuration for containers.

Last days we are migrating from LXC2 to LXC4 which used in our own orchestrator (aka stoneage LXD), been written long time ago in LXC1 ages, so this days we had to update conf-generator for low-level configurations.

Anyway, thanks for the help.
